File: Assets/Script/MyRoomState.cs
using Colyseus.Schema;

public class Player : Schema
{
    [Type(0, "number")] public float x = 0;
    [Type(1, "number")] public float y = 0;
}

public class MyRoomState : Schema
{
    [Type(0, "map", typeof(MapSchema<Player>))]
    public MapSchema<Player> players = new MapSchema<Player>();
}

File: my-colyseus/src/rooms/schema/MyRoomState.ts
import { Schema, type, MapSchema } from "@colyseus/schema";


export class Player extends Schema {
  @type("number") x = 0;
  @type("number") y = 0;
  @type("number") z = 0;
  @type("number") rotY = 0;
  @type("string") anim = "idle";
  @type("number") skin = 0;
  @type("boolean") ready = false;
  @type("number") clicks = 0;      // NEW: clicks during countdown
  @type("number") speed = 1.0;     // NEW: speed multiplier (default 1.0)
}

export class MyState extends Schema {
  @type({ map: Player })
  players = new MapSchema<Player>();

  @type("string") phase = "waiting";
  @type("number") countdown = 0;
}